### PR TITLE
Updated Jenkinsfiles to remove reliance on custom CF templates and fix `ssh` tests.

### PR DESCRIPTION
**Pull request by klueska**
Previously, we relied on custom cloud formation templates to launch the DC/OS clusters required by our CI jobs. This commit updates all of our integration test Jenkinsfiles to instead use dcos-launch in "onprem" mode with the proper configuration setup instead of running against these custom templates.

As part of this PR, it became obvious that some extra work was needed to get our CI system working properly for our various SSH tests with this change. The commits that fix those issues are included in this PR.

### CODE REVIEW

**Comment by klueska** [comment]
run all integration tests

**Comment by klueska** [comment]
rerun all integration tests

**Comment by klueska** [comment]
rerun all integration tests

**Comment by klueska** [comment]
rerun integration tests

**Comment by klueska** [comment]
rerun integration tests

**Comment by klueska** [comment]
rerun integration tests

**Comment by klueska** [comment]
run integration tests

**Comment by klueska** [comment]
rerun integration tests

**Comment by klueska** [comment]
rerun integration tests

**Comment by klueska** [comment]
rerun linux integration tests

**Comment by klueska** [comment]
rerun linux integration tests

**Comment by klueska** [comment]
rerun linux integration tests

File: cli/tests/integrations/test_node.py [hunk 1]
```diff
@@ -240,19 +240,24 @@ def test_node_ssh_slave_with_command():
...
+        args.extend(['--user', os.environ.get('CLI_TEST_SSH_USER')])
+
+    if os.environ.get('CLI_TEST_MASTER_PROXY') and \
+            '--master-proxy' not in args:
+        args.append('--master-proxy')
+
     cmd = ('ssh-agent /bin/bash -c "ssh-add {} 2> /dev/null && ' +
-           'dcos node ssh --option StrictHostKeyChecking=no {}"').format(
+           'dcos node ssh --option StrictHostKeyChecking=no ' +
+           '    --option ConnectTimeout=5 {}"').format(
```
Comment (bamarni) [inline cli/tests/integrations/test_node.py]: With a ccm cluster, 5 seconds is not enough when one is a few timezones away from the us-west region. I'm currently experiencing a test which fails most of the time (if not all the time), with 8 seconds it's passing most of the time.

Could we bump this to 8? That'd still be within the 10 seconds hard sleep in `ssh_output()`.

File: README.rst [hunk 2]
```diff
@@ -92,138 +92,172 @@ Running Tests
...
-    provider: aws
-    aws_region: us-west-2
-    template_parameters:
-        KeyName: default
-        AdminLocation: 0.0.0.0/0
-        PublicSlaveInstanceCount: 1
-        SlaveInstanceCount: 1
+*NOTE*: You **must** have your virtualenv *deactivated* in order to run the
+tests via the commands below. This is very important and often a point of
+much confusion.
```
Comment (bamarni) [inline README.rst]: Just curious, why is that?

File: cli/tests/integrations/test_node.py (hunk 1 above)
Comment (bamarni) [reply cli/tests/integrations/test_node.py]: I've just tried from your branch and my test seems to be much less flaky, as this is only the timeout to establish the ssh connection, there are still 5 other seconds after that for a potential command to run and finish. :+1:

**Comment by klueska** [comment]
run all integration tests

**Comment by klueska** [comment]
run all integration tests

**Comment by klueska** [comment]
run linux integration tests

**Comment by klueska** [comment]
run linux integration tests

**Comment by klueska** [comment]
run linux integration tests

**Comment by klueska** [comment]
run linux integration tests

**Comment by klueska** [comment]
rerun linux integration tests

File: README.rst [hunk 3]
```diff
@@ -92,138 +92,171 @@ Running Tests
...
+    os_name: cent-os-7
+    instance_type: m4.large
+    num_masters: 1
+    num_private_agents: 1
+    num_public_agents: 1
+    dcos_config:
+        cluster_name: DC/OS CLI Integration Tests
+        resolvers:
+            - 10.10.0.2
+        dns_search: mesos
```
Comment (bamarni) [inline README.rst]: We should adapt the readme too : `dns_search: us-west-2.compute.internal`

**Comment by klueska** [comment]
rerun all integration tests

File: README.rst (hunk 2 above)
Comment (bamarni) [reply README.rst]: @klueska : I had forgot about this one, could you explain why? How would integration tests will find the dcos executable for the current code?

**Pull request merged**